ui/src/pipeline/edit/transformation/MonacoDataProvider.vue

```diff
@@ -17,39 +17,56 @@
 <script lang="ts">
 import Vue from 'vue'
 import Component from 'vue-class-component'
-import { Prop } from 'vue-property-decorator'
+import { Watch } from 'vue-property-decorator'
 
 import MonacoEditor from 'vue-monaco'
+import { Action, State } from 'vuex-class'
+import { Data } from '@/datasource/datasource'
+
+const namespace = { namespace: 'transformation' }
 
 @Component({
   components: {
     MonacoEditor
   }
 })
 export default class MonacoDataProvider extends Vue {
-  @Prop() readonly value!: object
-
-  editorOptions = {
+  private editorOptions = {
     minimap: {
       enabled: false
     }
   }
 
-  object = this.value
-  text = this.formatJson(this.value)
+  /** from vuex module */
+  @State('data', namespace)
+  private data!: Data | null
+
+  /** from vuex module */
+  @Action('setDataAndSubmit', namespace)
+  private setDataAndSubmit!: (value: Data) => void
```

Comment:
we have this in `MonacoDataProvicer` and `CodeEditor`. So both components can trigger submitting data. When does happen what?

Reply:
We have `setDataAndSubmit` in `MonacoDataProvider`, which gets executed whenever we change the data in the text field.

The vuex function we have in `CodeEditor` is `setFunctionAndSubmit`, which changes the function.

Or did I misunderstand your comment?

Reply:
Was just a little confused. Thank's for the explanation!